src/chat_analyzer.py

```diff
@@ -158,15 +158,59 @@ def import_data(path_to_chatfile: str) -> List[Dict[str, Any]]:
     if isWa:
         print('Whatsapp chat recognized')
         f.seek(0)
+        #To check the dateformats
+        formatdmy = 1
+        formatmdy = 1
+        formatymd = 1
+        #To check that the date formats in the chat are in which format
+        #Can also check the time format here itself...
         for line in f:
             match = re.search(WMsg, line)
             if match:
+                try:
+                    datetime_object = datetime.strptime(match.groupdict()['date'], '%m/%d/%y')
+                except:
+                    formatmdy = 0
+                try:
+                    datetime_object = datetime.strptime(match.groupdict()['date'], '%d/%m/%y')
+                except:
+                    formatdmy = 0
+                try:
+                    datetime_object = datetime.strptime(match.groupdict()['date'], '%y/%m/%d')
+                except:
+                    formatymd = 0    
```

Comment:
So, instead of maintaining three identifiers for the different formats, what we can do is store the varying parameters, namely, datetime_object (already there), month, day, and year.